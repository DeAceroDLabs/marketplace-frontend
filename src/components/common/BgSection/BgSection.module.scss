@use "styles/colors";
.bgImage {
    height: 100%;
    object-fit: cover;
    object-position: 100% 30%;
    width: 100%;
}

.horizontal {
    height: 382px;
    width: 100%;
}

.vertical {
    height: 100%;
    width: 30%;
    max-width: 448px;
}

.content {
    height: inherit;
    position: absolute;
    text-align: center;
    width: inherit;
    top: 0%;
}

.left {
    left: 0%;
}

.right {
    right: 0%;
}

.primary {
    background-color: colors.$accent;
}