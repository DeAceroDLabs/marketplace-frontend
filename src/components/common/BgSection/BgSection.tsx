import styles from "./BgSection.module.scss";

interface BgSectionProps {
  bgImage?: string;
  orientation?: "vertical" | "horizontal";
  children?: React.ReactNode;
  color?: "primary" | "secondary";
  position?: "left" | "right" | "top" | "middle";
}

const BgSection: React.FC<BgSectionProps> = ({
  children,
  bgImage,
  position = "left",
  color = "transparent",
  orientation = "horizontal",
}) => {
  return (
    <div className={`${styles[orientation]}`}>
      {bgImage && <img src={bgImage} className={`${styles.bgImage}`} alt="" />}
      <div className={`${styles.content} ${styles[color]} ${styles[position]}`}>
        {children}
      </div>
    </div>
  );
};

export default BgSection;
