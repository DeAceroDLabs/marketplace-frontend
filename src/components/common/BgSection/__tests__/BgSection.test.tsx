import renderer from "react-test-renderer";
import BgSection from "../BgSection";

it("renders BgSection with no issue", () => {
  const component = renderer.create(<BgSection />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
