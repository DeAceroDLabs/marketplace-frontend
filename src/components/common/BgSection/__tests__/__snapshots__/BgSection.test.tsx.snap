// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BgSection with no issue 1`] = `
<div
  className="container horizontal"
>
  <div
    className="content"
  />
</div>
`;
